x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/DataFrameAnalysis.java

```diff
@@ -9,8 +9,13 @@
 import org.elasticsearch.common.xcontent.ToXContentObject;
 
 import java.util.Map;
+import java.util.Set;
 
 public interface DataFrameAnalysis extends ToXContentObject, NamedWriteable {
 
     Map<String, Object> getParams();
+
+    boolean supportsCategoricalFields();
```

Comment:
Could you please add at least one-liner comments to the interface methods? E.g. what does 'required' mean in getRequiredFields?

Reply:
Will do